akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/PredefinedFromStringUnmarshallers.scala

```diff
@@ -43,6 +45,15 @@ trait PredefinedFromStringUnmarshallers {
       }
     }
 
+  implicit val uuidFromStringUnmarshaller: Unmarshaller[String, UUID] =
+    Unmarshaller.strict { string ⇒
+      try UUID.fromString(string)
+      catch {
+        case e: IllegalArgumentException ⇒
+          throw new IllegalArgumentException(s"'$string' is not a valid UUID value", e)
```

Comment:
Does that wrapping improve the existing error message? In that case :+1:.

Reply:
@jrudolph I think it improves it in a sense that it makes it more consistent with all other predefined unmarshallers' errors.

Java's `UUID.fromString` exception error message is implemented like this:

 `"Invalid UUID string: "+name`.

Now here are other predefined from string unmarshallers' errors in this file:

`s"'$x' is not a valid Boolean value"`
`s"'$value' is not a valid 8-bit hexadecimal integer value"`
`s"'$value' is not a valid 8-bit signed integer value"`

It's arguable whether it's worth wrapping exception for this, but I voted for consistency (and prettiness)

Reply:
I see, thanks for the explanation.